### PR TITLE
Switch to and update GitHub Actions

### PR DESCRIPTION
**Pull request by jmert**
Mostly copied from JuliaIO/MAT.jl's GitHub Actions setup.

### CODE REVIEW

**Review by musm — APPROVED** [review]

**Comment by jmert** [comment]
Julia v1.0 failures look to be known — the old Travis CI script excluded the JLDArchives tests on 1.0, and that appears to be just because the Pkg API doesn't like the download method. We're about to bump to minimum Julia 1.3, so that's fine.

The Windows failures on 1.5 and nightly are interesting, but probably worth debugging after merging #280 since the large number of changes in HDF5 v0.14 might have already fixed some subtle issue.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        version:
+          - '1.0'
+          - '1'
+          - 'nightly'
+        os:
+          - ubuntu-latest
+          - macOS-latest
+          - windows-latest
+        arch:
+          - x64
```
Comment (musm) [inline .github/workflows/CI.yml:21]: Are we not test x86 ?  (Might be worth to add at least for one of the platforms, usually windows x86 is the most finicky).

Comment (musm) [reply .github/workflows/CI.yml:21]: Although neither MAT.jl tests on x86 anymore.

Comment (jmert) [reply .github/workflows/CI.yml:21]: I can copy in the matrix from HDF5 to #280 since I have to rebase on top of this now.

**Pull request merged**